core/src/main/scala/cats/data/Binested.scala

```diff
@@ -0,0 +1,93 @@
+package cats
+package data
+
+import cats.arrow._
+
+/** Compose a two-slot type constructor `F[_, _]` with two single-slot type constructors
+ *  `G[_]` and `H[_]`, resulting in a two-slot type constructor with respect to the inner types.
+ *  For example, `List` and `Option` both have `Functor` instances, and `Either` has a
+ *  `Bifunctor` instance. Therefore, `Binested[Either, List, Option, ?, ?]` has a `Bifunctor`
+ *  instance as well:
+ *
+ * {{{
+ * scala> import cats.Bifunctor
+ * scala> import cats.data.Binested
+ * scala> import cats.implicits._
+ * scala> val eitherListOption: Either[List[Int], Option[String]] = Right(Some("cats"))
+ * scala> val f: Int => String = _.toString
+ * scala> val g: String => String = _ + "-bifunctor"
+ * scala> val binested = Binested(eitherListOption)
+ * scala> val bimapped = Bifunctor[Binested[Either, List, Option, ?, ?]].bimap(binested)(f, g).value
+ * res0: Either[List[String], Option[String]] = Right(Some("cats-bifunctor"))
+ * }}}
+ */
+final case class Binested[F[_, _], G[_], H[_], A, B](value: F[G[A], H[B]])
+
+object Binested extends BinestedInstances
+
+trait BinestedInstances extends BinestedInstances0 {
+  implicit def catsDataEqForBinested[F[_, _], G[_], H[_], A, B](
+    implicit F: Eq[F[G[A], H[B]]]): Eq[Binested[F, G, H, A, B]] =
+    Eq.by(_.value)
+
+  implicit def catsDataBifunctorForBinested[F[_, _], G[_], H[_]](
+    implicit F: Bifunctor[F], G: Functor[G], H: Functor[H]): Bifunctor[Binested[F, G, H, ?, ?]] =
+    new Bifunctor[Binested[F, G, H, ?, ?]] {
+      def bimap[A, B, C, D](fab: Binested[F, G, H, A, B])(f: A => C, g: B => D): Binested[F, G, H, C, D] =
+        Binested(F.bimap(fab.value)(G.map(_)(f), H.map(_)(g)))
+    }
+
+  implicit def catsDataProfunctorForBinested[F[_, _], G[_], H[_]](
+    implicit F: Profunctor[F], G: Functor[G], H: Functor[H]): Profunctor[Binested[F, G, H, ?, ?]] =
+    new Profunctor[Binested[F, G, H, ?, ?]] {
+      def dimap[A, B, C, D](fab: Binested[F, G, H, A, B])(f: C => A)(g: B => D): Binested[F, G, H, C, D] =
+        Binested(F.dimap(fab.value)(G.map(_: G[C])(f))(H.map(_)(g)))
+    }
+
+  implicit def catsDataBifoldableForBinested[F[_, _], G[_], H[_]](
+    implicit F0: Bifoldable[F], G0: Foldable[G], H0: Foldable[H]): Bifoldable[Binested[F, G, H, ?, ?]] =
+    new BinestedBifoldable[F, G, H] {
+      override implicit def F: Bifoldable[F] = F0
+      override implicit def G: Foldable[G] = G0
+      override implicit def H: Foldable[H] = H0
+    }
+}
+
+trait BinestedInstances0 {
+  implicit def catsDataBitraverseForBinested[F[_, _], G[_], H[_]](
```

Comment:
`Bitraverse` should take higher priority than `Bifoldable` and `Bifunctor`
https://typelevel.org/cats/guidelines.html#a-idimplicit-priority-hrefimplicit-prioritya-implicit-instance-priority

Reply:
Right, thanks